lamindb/curators/__init__.py

```diff
@@ -703,22 +730,186 @@ def save_artifact(
         )
 
 
+class SpatialDataCurator(SlotsCurator):
+    # the example in the docstring is tested in test_curators_quickstart_example
+    """Curator for a SpatialData object.
+
+    See also :class:`~lamindb.Curator` and :class:`~lamindb.Schema`.
+
+    .. versionadded:: 1.3.0
+
+    Args:
+        dataset: The SpatialData-like object to validate & annotate.
+        schema: A `Schema` object that defines the validation constraints.
+
+    Example::
+
+        import lamindb as ln
+        import bionty as bt
+
+        # define sample schema
+        sample_schema = ln.Schema(
+            name="blobs_sample_level_metadata",
+            features=[
+                ln.Feature(name="assay", dtype=bt.ExperimentalFactor).save(),
+                ln.Feature(name="disease", dtype=bt.Disease).save(),
+                ln.Feature(name="development_stage", dtype=bt.DevelopmentalStage).save(),
+            ],
+            coerce_dtype=True
+        ).save()
+
+        # define table obs schema
+        blobs_obs_schema = ln.Schema(
+            name="blobs_obs_level_metadata",
+            features=[
+                ln.Feature(name="sample_region", dtype="cat[ULabel]").save(),
+            ],
+            coerce_dtype=True
+        ).save()
+
+        # define table var schema
+        blobs_var_schema = ln.Schema(
+            name="blobs_var_schema",
+            itype=bt.Gene.ensembl_gene_id,
+            dtype=int
+        ).save()
+
+        # define composite schema
+        spatialdata_schema = ln.Schema(
+            name="blobs_spatialdata_schema",
+            otype="SpatialData",
+            components={
+                "sample": sample_schema,
+                "table:obs": blobs_obs_schema,
+                "table:var": blobs_var_schema,
+        }).save()
+
+        # curate a SpatialData
+        spatialdata = ln.core.datasets.spatialdata_blobs()
+        curator = ln.curators.SpatialDataCurator(spatialdata, spatialdata_schema)
+        try:
+            curator.validate()
+        except ln.errors.ValidationError as error:
+            print(error)
+
+        # add new categorical values and standardize
+        curator.slots["sample"].cat.add_new_from(key="developmental_stage")
```

Comment:
We want to make sure users don't blindly call `add_new_from`, so I'd remove these calls from the examples and register new records at the beginning, similar to AnnData and MuData examples.

We have an example of `add_new_from` in `curate-df`, so users can still find it.

Reply:
Can `sample_region` be of dtype `str` if it's not meant to have controlled vocabulary?

Reply:
1. Removed the calls.
2. Can't be validated because I came up with this term (by design). Using a name that exists now.
3. Yes, to make the example simpler. Done.